lld/MachO/ConcatOutputSection.cpp

```diff
@@ -116,6 +116,30 @@ void ConcatOutputSection::addInput(ConcatInputSection *input) {
 
 DenseMap<Symbol *, ThunkInfo> lld::macho::thunkMap;
 
+// Returns the target Symbol that a relocation refers to.
+// A Reloc can refer to either a Symbol directly, or to an InputSection.
+// For InputSection referents, we return the first Symbol at offset 0.
+// This conversion is necessary because the thunk generation algorithm
+// can only handle Symbols as branch targets, not InputSections.
+static Symbol *getReferentSymbol(const Reloc &r) {
+  if (auto *sym = r.referent.dyn_cast<Symbol *>()) {
+    return sym;
+  } else if (auto *isec = r.referent.dyn_cast<InputSection *>()) {
```

Comment:
Likewise, may I suggest the following?

```suggestion
  } else if (auto *isec = llvm::dyn_cast<InputSection *>(r.referent)) {
```

Reply:
Also, don't use `else` after a `return`
https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return